File: CleverFerretV2/core/network/src/main/java/com/cleverferret/v2/core/network/resilience/NonIdempotentRetryPolicy.java
package com.cleverferret.v2.core.network.resilience;

import com.cleverferret.v2.core.common.error.AppError;

public final class NonIdempotentRetryPolicy implements RetryPolicy {
    @Override
    public boolean shouldRetry(AppError error, int attempt) {
        return attempt < 2 && error instanceof AppError.TransportFailure transportFailure && transportFailure.timeout();
    }

    @Override
    public long delayMillis(int attempt) {
        return 150L;
    }

    @Override
    public int maxAttempts() {
        return 2;
    }
}

File: CleverFerretV2/core/network/src/main/java/com/cleverferret/v2/core/network/resilience/NonIdempotentRetryPolicy.kt
package com.cleverferret.v2.core.network.resilience

import com.cleverferret.v2.core.common.error.AppError

class NonIdempotentRetryPolicy : RetryPolicy {
    override fun shouldRetry(error: AppError, attempt: Int): Boolean =
        attempt < 2 && error is AppError.TransportFailure && error.timeout

    override fun delayMillis(attempt: Int): Long = 150L
    override fun maxAttempts(): Int = 2
}
